server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -216,10 +209,19 @@ func (r *Runner) Apply(ctx workflow.Context, root *terraform.LocalRoot, serverUR
 func (r *Runner) Run(ctx workflow.Context) error {
 	var response *activities.GetWorkerInfoResponse
 	err := workflow.ExecuteActivity(ctx, r.TerraformActivities.GetWorkerInfo).Get(ctx, &response)
-
 	if err != nil {
 		return errors.Wrap(err, "getting worker info")
 	}
+	//we have to reset all activity options since WithActivityOptions doesn't append
+	options := workflow.ActivityOptions{
+		ScheduleToCloseTimeout: ScheduleToCloseTimeout,
+		HeartbeatTimeout:       HeartBeatTimeout,
+		RetryPolicy: &temporal.RetryPolicy{
+			NonRetryableErrorTypes: []string{TerraformClientError},
+		},
+		TaskQueue: response.TaskQueue,
+	}
+	ctx = workflow.WithActivityOptions(ctx, options)
```

Comment:
Because `GetWorkerInfo` happens in the terraform workflow, we have to register the Terraform workflow and activities (really just this one activity) to the new terraformw worker in server/neptune/temporalworker/server.go. If this is an issue, I believe one alternative could be to move this activity call into the parent deploy workflow and pass this information into the terraform workflow when executing it.

Reply:
It shouldn't matter, we only care if the terraform workflow stateful activities all exist in the same instance .